### PR TITLE
[GLUTEN-1393] Add ValueStreamNode operator

### PR DESCRIPTION
**Pull request by jinchengchenghh**

### CODE REVIEW

**Comment by jinchengchenghh** [comment]
Can you help review this one? Thanks! @rui-mo 
Gluten: https://github.com/oap-project/gluten/pull/1397

File: velox/exec/LocalPlanner.cpp [hunk 1]
```diff
@@ -398,6 +402,11 @@ std::shared_ptr<Driver> DriverFactory::createDriver(
         auto valuesNode =
             std::dynamic_pointer_cast<const core::ValuesNode>(planNode)) {
       operators.push_back(std::make_unique<Values>(id, ctx.get(), valuesNode));
+    } else if (
+        auto arrowStreamNode =
```
Comment (rui-mo) [inline velox/exec/LocalPlanner.cpp:406]: arrowStreamNode -> valueStreamNode?

File: velox/exec/ValueStream.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          driverCtx,
+          valueStreamNode->outputType(),
+          operatorId,
+          valueStreamNode->id(),
+          "ValueStream") {
+  valueStream_ = valueStreamNode->rowVectorStream();
+}
+
+RowVectorPtr ValueStream::getOutput() {
+  TestValue::adjust("facebook::velox::exec::ValueStream::getOutput", this);
```
Comment (rui-mo) [inline velox/exec/ValueStream.cpp:37]: Why do we need this?

File: velox/exec/ValueStream.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  RowVectorPtr getOutput() override;
+
+  BlockingReason isBlocked(ContinueFuture* /* unused */) override {
+    return BlockingReason::kNotBlocked;
+  }
+
+  void noMoreInput() override {
+    Operator::noMoreInput();
+    close();
```
Comment (rui-mo) [inline velox/exec/ValueStream.h:36]: Why close is called in noMoreInput?

File: velox/exec/ValueStream.cpp (hunk 2 above)
Comment (jinchengchenghh) [reply velox/exec/ValueStream.cpp:37]: Just copy from https://github.com/oap-project/velox/blob/main/velox/exec/Values.cpp#L45
I will remove it

File: velox/exec/ValueStream.h (hunk 3 above)
Comment (jinchengchenghh) [reply velox/exec/ValueStream.h:36]: Copy from here https://github.com/oap-project/velox/blob/main/velox/exec/Values.h#L36
Maybe it is not essential here?

Comment (jinchengchenghh) [reply velox/exec/ValueStream.h]: Removed

**Comment by rui-mo** [comment]
Can we test this PR on Jenkins to make sure the funtionality is alright?

**Comment by jinchengchenghh** [comment]
Yes, I have tested TPCH and TPCDS in jenkins, the performance is same as before, You can search the test report in your mail box.

File: velox/exec/ValueStream.cpp (hunk 2 above)
Comment (rui-mo) [reply velox/exec/ValueStream.cpp:37]: Looks this is for parallel execution, and we can remove it.

**Comment by rui-mo** [comment]
> Yes, I have tested TPCH and TPCDS in jenkins, the performance is same as before, You can search gluten PR `1397` to get the test report in your mail box.

Do we need to test again after PR change?

**Comment by jinchengchenghh** [comment]
> > Yes, I have tested TPCH and TPCDS in jenkins, the performance is same as before, You can search gluten PR `1397` to get the test report in your mail box.
> 
> Do we need to test again after PR change?

I have triggered gluten CI after code change.
The change should not affect performance, anyway, I will trigger in jenkins again.

**Comment by rui-mo** [comment]
Thanks!

**Review by rui-mo — APPROVED** [review]

**Pull request merged**